### PR TITLE
[Fix] Fix entropy calculation without sample packing

### PR DESCRIPTION
**Pull request by SumanthRH**
# What does this PR do?

Fixes entropy metric calculation when sample packing is not used

TODO:
- [x] Validate with seq parallel

### CODE REVIEW

File: skyrl-train/tests/cpu/utils/test_torch_utils.py [hunk 1]
```diff
@@ -52,3 +52,64 @@ def test_chunked_entropy_from_logits():
...
+    except ValueError as e:
+        assert "does not match logits shape" in str(e)
+
+    # Wrong shape: different sequence length
+    wrong_mask_seqlen = torch.ones(2, 4, dtype=torch.float32)
+    try:
+        chunked_entropy_from_logits(logits, attention_mask=wrong_mask_seqlen)
+        assert False, "Should have raised ValueError for wrong sequence length"
+    except ValueError as e:
+        assert "does not match logits shape" in str(e)
```
[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl-train/tests/cpu/utils/test_torch_utils.py]: yea

File: skyrl-train/skyrl_train/workers/worker.py [hunk 2]
```diff
@@ -755,13 +754,9 @@ def training_step(self, experience: Experience, global_step, local_step, accumul
             )
         # entropy
         with torch.no_grad():
-            if self.cfg.trainer.use_sample_packing:
-                # batch_size, seqlen
-                entropy_BS = output["entropy"]
-                entropy_BS = entropy_BS[:, -num_actions - 1 : -1]
-            else:
-                action_logits = output["logits"][:, -num_actions - 1 : -1, :]
-                entropy_BS = chunked_entropy_from_logits(action_logits, requires_grad=False)
```
Comment (SumanthRH) [inline skyrl-train/skyrl_train/workers/worker.py:764]: Can remove this code block after #480

File: skyrl-train/skyrl_train/workers/worker.py [hunk 3]
```diff
@@ -755,13 +754,9 @@ def training_step(self, experience: Experience, global_step, local_step, accumul
...
+            # batch_size, seqlen
+            entropy_BS = output["entropy"]
+            entropy_BS = entropy_BS[:, -num_actions - 1 : -1]
```
[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl-train/skyrl_train/workers/worker.py:759]: Hmm actually entropy value is consistent for with sample packing and without sample packing - because packing always happens internally in the modelling code

[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl-train/skyrl_train/workers/worker.py:759]: So the previous comptation itself was wrong

[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl-train/skyrl_train/workers/worker.py:759]: Fixed now @gemini-code-assist
